### PR TITLE
Added ide generated files to gitignore.

### PR DESCRIPTION
**Pull request by HardieBoeve**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,13 @@
+.buildpath
+.DS_Store
+.idea
+.project
+.settings/
+.*.sw*
+.*.un~
+nbproject
+tmp/
+
```
Comment (samsonasik) [inline .gitignore]: remove the blank line

**Comment by weierophinney** [comment]
Shouldn't these be in individual users' global gitignore rules, and not in the repository?

I have, in my `$HOME/.gitconfig`, the setting `core.excludesfile` set to the value of `$HOME/.gitignore-global`; that file contains exclusions for IDE-generated files. I recommend all contributors do the same.

**Comment by Maks3w** [comment]
I don't remind to see a PR with foreign files added. I think the level of mature is enough so there is no need to exclude those files.

**Comment by HardieBoeve** [comment]
Ok, closed as there are better alternatives to exclude these files.

**Pull request closed without merging**